modules/installation-and-upgrade/pages/install-vm.adoc

```diff
@@ -46,19 +43,150 @@ Find the {minimalsles} image at https://download.suse.com/.
+
+```
+openssl passwd -6
+```
+
+The command creates a hash of the password you chose. Use this hash as the value of the [systemitem]``password_hash`` attribute.
+
+The users attribute must contain at least one [systemitem]``name`` attribute. [systemitem]``ssh_authorized_keys`` is a list of ssh keys for the user. 
+
+Prepare required configuration using following commands:
+
+```
+$ mkdir -p root/ignition
+$ cat << EOF > root/ignition/config.ign
+{
+  "ignition": {
+    "version": "3.2.0"
+  },
+  "passwd": {
+    "users": [
+      {
+        "name": "root",
+        "passwordHash": "$2a$10$qV298UV11u9lCFDjpHpCUe1cErBiVR.G3shukxs3.2PAO1xhJWs0K"
+      }
+    ]
+  }
+}
+
+EOF
+```
+
+Prepare [systemitem]``Ingition`` ISO file using the command:
+```
+mkisofs -full-iso9660-filenames -o suma_ignition.iso -V ignition root
+```
+
+Attach created [path]``suma_ignition.iso`` file as a volume to the creating virtual machine. This particular example is setting [systemitem]``root`` password to `linux`.
+
+For more information about [systemitem]``Ignition``, see https://documentation.suse.com/sle-micro/5.4/single-html/SLE-Micro-deployment/#cha-images-ignition
+
+[[sumavm.cloud_init]]
+== {productname} basic configuration using Cloud Init disk
+
+[systemitem]``Cloud Init`` is a provisioning tool that enables you to configure a system according to your specification on the first boot.
+When the system is booted for the first time, [systemitem]``Cloud Init service`` is loaded and searches for a configuration file within a specific directory (on a USB flash disk, or you can provide a URL).
+
+[systemitem]``Cloud Init`` uses few configuration files in the YAML format. Used files are named [path]``meta-data``, [path]``network-config`` and [path]``user-data``.
+
+[systemitem]``Cloud Init`` allows numerous sources where to store configuration data. In this guide we use local iso image with volume id [systemitem]``cidata`` as a source. The directory structure must look as follows:
+
+```
+<root directory>
+└── meta-data
+└── network-config
+└── user-data
+```
+
+If you intend to configure a QEMU/KVM virtual machine, provide the path to the [path]``config.ign`` as an attribute of the qemu command. For example:
+
+```
+-fw_cfg name=opt/com.coreos/config,file=PATH_TO_config.ign
+```
+
+The [systemitem]``Cloud Init`` allows many management options. For a complete specification, refer to Cloud Init specification (https://cloudinit.readthedocs.io/en/latest/index.html).
+
+=== Setup root password using Cloud Init
+
+You need to hash the root password, for example, by using the openssl command:
+
+```
+openssl passwd -6
+```
 
+The command creates a hash of the password you chose. Use this hash as the value of the [systemitem]``password`` attribute.
 
-[[minimmal.kvm.settings]]
-== Minimal SLES KVM Settings
+Prepare needed configuration files using following commands:
+```
+$ touch network-config
+$ touch meta-data
+$ cat << EOF > user-data
```

Comment:
```suggestion
Create a file named user-data with the following content:
```
this is the same issue as ignition, the terminal mangles the hash, you need to escape special characters etc